### DISCUSSION TITLE
More forgiving errors during dev with missing filters

### DISCUSSION THREAD

**Discussion by LeaVerou** [11ty/eleventy#4005] [Enhancement Queue]
I’ve been using filters _extensively_. Most of my 11ty projects have an import like:

```js
import * as filters from "./filters/index.js";
```

and a mass-add statement in the config:

```js
for (let name in filters) {
	config.addFilter(name, filters[name]);
}
```

If a filter is invalid, 11ty currently throws (and the error message code path is broken, see #3385).
I’d argue that this is not a reason for complete and total failure — a warning should suffice. Something like "Filter `foo` ignored, as it is not a function". 

However, then if filter `foo` is used anywhere, _that_ would throw. But should it really? 

As part of debugging, I often comment out plugins. But that means now their filters produce errors! So temporarily disabling a plugin involves tediously adding stubs like 

```js
config.addFilter("foo", content => content);
```

What if 11ty could do this for any unrecognized filter, and just printed out a warning? Fault tolerance and robustness are generally good API design principles to follow. The main case where you _don't_ want to fail gracefully is when making the wrong assumption could produce data loss (see #3373), but that doesn’t seem to be the case here.

**Comment by zachleat** [comment]
I bet we could get away with a template-syntax independent implementation of this feature using Proxy. I’d imagine folks might want it to operate this way during dev but not in a full production build, too.

**Comment by zachleat** [comment]
> By default this repo closes enhancement request issues and places them in a queue for folks to upvote. We do this to avoid a large build-up of open issues. We got this idea from the [lodash project](https://twitter.com/samselikoff/status/991395669016436736).

> Don’t forget to upvote the first comment with 👍 to register your vote! You can view [the current enhancement backlog](https://github.com/11ty/eleventy/issues?utf8=%E2%9C%93&q=label%3Aneeds-votes+sort%3Areactions-%2B1-desc+).
